### PR TITLE
Create Google Summer of Code 2023 proposal for moja global

### PR DESCRIPTION
**Pull request by HarshCasper**

### CODE REVIEW

**Comment by aornugent** [comment]
These look great @HarshCasper - can you please verify that Deval and Anthony are available to mentor the Deforestation UI and FLINT Reporting projects respectively.

**Comment by WhiteWolf47** [comment]
hey, is moja global participating in gsoc 2023? I would really love to contribute (im new to moja global), can you please link your communication channels here?

File: google-summer-of-code/GSOC-2023-Ideas.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+
+The FLINT is based on over 20 years of experience building and operating measurement, reporting, and verification (MRV) tools in Australia and Canada. The FLINT provides the capability to quickly establish an operational MRV system responding to policymaking and planning needs. The FLINT has been designed to achieve the required accuracy through step-by-step improvements.
+
+New FLINT users are expected to follow the IPCC guidelines to develop increasingly accurate and robust systems for greenhouse gas accounting. Users are supported by moja global, a vibrant developer community, and philanthropic partners like the UNFCCC and IUCN. We aim to grow our institutional users and require excellent communication, education, and documentation across various topics, including science, policy, and software development.
```
Comment (aornugent) [inline google-summer-of-code/GSOC-2023-Ideas.md:9]: Remove IUCN please.

File: google-summer-of-code/GSOC-2023-Ideas.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+
+- Land Sector Datasets: https://github.com/moja-global/Land_Sector_Datasets
+- FLINT.Cloud: https://github.com/moja-global/FLINT.Cloud
+- MongoDB Python connection: https://www.mongodb.com/languages/python
+
+### First steps
+
+The best way to get started is to join our Slack community: https://join.slack.com/t/mojaglobal/shared_invite/zt-o6ta1ug0-rVLjAo460~d7JbZ~HpFFtw
+
+Please introduce yourself in `#cloud` - we'd love to know where you come from and what you're interested in. Then proceed to `#documentation` and the [FLINT Handbook](https://moja-global.github.io/Handbook/). If you'd like to install FLINT please review the instructions at [docs.moja.global](https://docs.moja.global) and as for help in `#installation` support. We have some demonstration guides in the FLINTcloud and GCBM.Belize that you can follow to better understand the types of models we plan on using.
```
Comment (aornugent) [inline google-summer-of-code/GSOC-2023-Ideas.md:63]: Please double check these channels are correct.

File: google-summer-of-code/GSOC-2023-Ideas.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+
+During this project, the developer will work with the Technical Steering Committee (TSC) on contributing to FLINT.Reporting while getting hands-on experience with Spring Boot, Angular, PostgreSQL, and Docker while understanding FLINT.Reporting in depth. Experience with Spring Boot and PostgreSQL is preferred. Experience with Angular and Docker would be helpful. Understanding FLINT would be preferred but optional.
+
+### Resources
+
+- FLINT.Reporting: ​​https://docs.moja.global/projects/flint-reporting/en/latest/index.html
+
+### First steps
+
+Get FLINT.Reporting up and running on your local machine by following the docs and understanding the various microservices and software components. After setup and initial discovery, share your findings on the `#reporting-tool` channel.
```
Comment (aornugent) [inline google-summer-of-code/GSOC-2023-Ideas.md:198]: Please update channel name

File: google-summer-of-code/GSOC-2023-Ideas.md [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+
+### Technical skills
+
+Experience designing and implementing Web Applications while managing end-to-end client and server-side development is a must! Understanding of a popular Python web framework (FastAPI, Flask, Django) is needed. The choice of the UI library/tools is flexible and will be based on the consensus of the mentor and the mentee. Understanding how a complete UI stack is set up would help the project be finished smoothly. Knowledge of the formats in which large geographical images are stored/rendered is preferred but optional.
+
+### First steps
+
+Please get started by reading and understanding the problem statement and our approach in this [publically available case study](https://community.moja.global/case-studies/powering-deforestation-prediction-using-artificial-intelligence). Try to understand the case study, the use-cases and how you can build an application around it. Share your thoughts on the `#machine-learning` channel and start working on the project's rough architecture and system design.
```
Comment (aornugent) [inline google-summer-of-code/GSOC-2023-Ideas.md:279]: Please update channel name

File: google-summer-of-code/GSOC-2023-Ideas.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+
+Currently, the FLINT.Cloud offers two microservices: FLINT.Example REST API and GCBM REST API. The current implementation follows a functional programming paradigm that is difficult to scale. We have often faced problems implementing new API endpoints and following a consistent schema that would enable us to version our APIs and work on new features. We also wish to switch to an asynchronous, event-driven mode of executing the simulations to ensure a more efficient API layout and an increased processing velocity.
+
+The goal of this project is to implement an object-oriented refactor for FLINT.Cloud, including the addition of `Flask-RESTful` to extend the project modularity. The project should also involve the investigation and implementation of asynchronous task processing to make the carbon simulations more efficient. The current scope of work also entails implementing the unit, integration, and E2E tests to cover the new functional implementation and scale aspects.
```
Comment (aornugent) [inline google-summer-of-code/GSOC-2023-Ideas.md:289]: This is a great description - @SanjaySinghRajpoot also wanted to evaluate FastAPI, so perhaps we don't need to tie ourselves to FlaskRESTful at this stage?

File: google-summer-of-code/GSOC-2023-Ideas.md [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+- Investigate and implement asynchronous task processing to make the carbon simulations more efficient for our users and benchmark the performance of the new implementation.
+- Present a proof-of-concept for event-driven processing of the carbon simulations, to make the microservices protocol-agnostic and document the new implementation.
+
+### Technical skills
+
+Experience with Python, Flask, and Object-oriented programming is preferable. Understanding Docker and various engineering aspects, such as design, development, testing, and packaging, is useful. Experience with asynchronous task processing and event-driven programming is a plus. Knowledge of the FLINT ecosystem, including various moja global projects, carbon models, and GCBM, is preferred but optional.
+
+### First steps
+
+Introduce yourself on Slack in the `#cloud` channel. Download the FLINT.Cloud GCBM container and re-create the GCBM Demo Run. Share your findings on Slack under `#cloud`. Next, create a system design for the new implementation focusing on improving the API performance, and share it on Slack.
```
Comment (aornugent) [inline google-summer-of-code/GSOC-2023-Ideas.md:317]: Please check channel names

File: google-summer-of-code/GSOC-2023-Ideas.md [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+
+## Idea: Improve FLINT-UI's performance and accessibility for better user-experience
+
+[FLINT-UI](https://github.com/moja-global/flint-ui)  provides an intuitive way for new users to explore some preconfigured FLINT modules, including the  [Generic Carbon Budget Model (GCBM)](https://community.moja.global/docs/GCBM/GCBM). The FLINT-UI client is written as a web application and can be used in a local or remote environment. [moja global's UI library](https://github.com/moja-global/ui-library) provides a consistent and easy-to-use library to improve our design & development workflow while satisfying our performance and accessibility requirements. Currently, the FLINT-UI consumes the various components provided by the UI library, and we wish to expand it further to accommodate more use cases.
```
Comment (aornugent) [inline google-summer-of-code/GSOC-2023-Ideas.md:359]: This is missing the ## Abstract heading

File: google-summer-of-code/GSOC-2023-Ideas.md [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+- Coordinating with the Documentation Working group to use the new FlINT Handbook to mitigate common usage questions and improve onboarding for new users.
+
+### Technical skills
+
+During this project, the developer will be working with the UI working group under the Technical Steering Committee (TSC) on contributing to FLINT-UI while getting hands-on experience on working with VueJS, Web accessibility & performance, Docker and testing. Experience with VueJS and JavaScript is preferred. Experience with testing and automation would be useful. Understanding of FLINT & GCBM would be preferred but not necessary.
+
+### First steps
+
+Introduce yourself on Slack in the `#user-interface` channel and review the existing preconfigured FLINT modules on FLINT-UI. After running those modules, share your inferences as a small document to help us review your understanding of the project. Next, create a small contribution to the FLINT-UI project and ask mentors for feedback. Review the UI library and identify scopes for improvement to make them more accessible and performant.
```
Comment (aornugent) [inline google-summer-of-code/GSOC-2023-Ideas.md:391]: Please check channel names.

**Review by aornugent — APPROVED** [review]
Amazing work! Thanks @HarshCasper!

File: google-summer-of-code/GSOC-2023-Ideas.md (hunk 5 above)
Comment (HarshCasper) [reply google-summer-of-code/GSOC-2023-Ideas.md:289]: Sure, I will add this to the scope.

**Pull request merged**